nodejs/device/src/index.ts

```diff
@@ -418,6 +444,11 @@ class DeviceApp {
   };
 
   private async startTestOperations() {
+    logger.info("Starting thief reported property sending.");
+    this.thiefPropertySendingInterval = setInterval(
```

Comment:
Not related to this change, but in the medium term, we still need to figure out if there is a better way to return failure from an interval function.

Reply:
Yep, I have a list of small design tweaks that I want to make and that's on it. 

For errors related to metrics and logging, would we even want to do anything other than logging the error and moving on? They aren't the things that we are trying to test, so I'm not sure we want them to cause the program to terminate.

Reply:
Also, the fact that built-in JS promises don't have cancellation built in is kinda frustrating, because that would work wonderfully for this sort of thing.